File: 2022/02/rock-paper-scissors.jl
data = readlines("input.txt")

moves = [[Int(i[1]) - Int('A') + 1, Int(i[3]) - Int('X') + 1] for i in data]
them_vs_us = [1 2 0; 0 1 2; 2 0 1]

points = sum([them_vs_us[i[1], i[2]] for i in moves]) * 3 + sum([i[2] for i in moves])
@show points

them_vs_us = reverse(them_vs_us, dims=1) .+ 1

points = sum([them_vs_us[i[1], i[2]] for i in moves]) + (sum([i[2] - 1 for i in moves]) * 3)
@show points

File: 2022/02/rock-paper-scissors.py
with open('input.txt', 'r') as f:
    cheat_sheet = f.readlines()

them_vs_us = [
    [1, 2, 0],
    [0, 1, 2],
    [2, 0, 1]
]

# Part One
total = 0
for line in cheat_sheet:
    them, me = (ord(line[0]) - ord('A'), ord(line[2]) - ord('X'))
    result = them_vs_us[them][me] * 3
    total += result + (me + 1)

print(total)

# Part Two
total = 0
for line in cheat_sheet:
    them, result = (ord(line[0]) - ord('A'), ord(line[2]) - ord('X'))
    me = them_vs_us[::-1][them][result] + 1
    total += ((result * 3) + me)

print(total)
